### DISCUSSION TITLE
Import csv file with columns of type text, but values are numbers starting with zeros

### DISCUSSION THREAD

**Discussion by mztmoraes** [supabase/supabase#15685] [Questions]
Hey guys,
I have a csv file with columns that the values of these columns are numbers with zeros at the beginning. For example: 0001, 0002, 0003, etc.
When I open the csv file in excel or notepad++ for verification, I see that the values appear normally as 0001, 0002, 0003.

When importing to Supabase with the csv import function, even selecting the column type to be TEXT, the values appear without leading zeros, like this: 1, 2, 3, 4.

I have tried all import ways.

1. by excel, in several ways. Formatting in different ways.
2. I created a txt file in notepad++, manually, with just a few rows. I saved it in txt and renamed it to csv. I imported into Supabase and nothing.
3. I tested it in two ways: creating the table before with the fields as TEXT type and importing the csv in the table. And the other way was to create the table directly by importing the CSV.

Is this a Supabase limitation or a problem? 

in the attached image, note that the table was created and I tried several models. That was the last table. So I tried manually entering the information (indicated by the arrow) and the table shows the value with leading zeros (001).

Does anyone have any ideas to help me?

![image](https://github.com/supabase/supabase/assets/18100786/574d6821-50c0-4ea1-b954-21d3199f153e)

**Comment by jhaus** [comment]
Having the same issue importing a CSV file with zip codes that include leading zeros. In addition to the original posters attempts, I have also:
- wrapped the zip code strings in double quotes
- used various types by creating the table first in the SQL editor, including: numeric(5) / char(5) / text / varchar
- verified the format in VS Code before importing
- verified the leading zero is displayed in the "Preview data to be imported" view

In every circumstance, the leading zeros are removed upon import of the CSV file.
